<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SubjectDialog</class>
 <widget class="QDialog" name="SubjectDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>500</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Create New Subject</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="main_layout">
   <property name="margin">
    <number>20</number>
   </property>
   <property name="spacing">
    <number>15</number>
   </property>
   <item>
    <widget class="QLabel" name="header_label">
     <property name="text">
      <string>Subject Information</string>
     </property>
     <property name="styleSheet">
      <string>QLabel { font-size: 16px; font-weight: bold; color: #2c3e50; }</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </item>

   <item>
    <widget class="QGroupBox" name="basic_info_group">
     <property name="title">
      <string>Basic Information</string>
     </property>
     <layout class="QFormLayout" name="basic_info_layout">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
      </property>

      <item row="0" column="0">
       <widget class="QLabel" name="subject_id_label">
        <property name="text">
         <string>Subject ID: *</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="subject_id_edit">
        <property name="placeholderText">
         <string>Enter unique subject identifier</string>
        </property>
        <property name="toolTip">
         <string>Unique identifier for the subject (required)</string>
        </property>
       </widget>
      </item>

      <item row="1" column="0">
       <widget class="QLabel" name="species_label">
        <property name="text">
         <string>Species:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="species_combo">
        <property name="toolTip">
         <string>Select the subject species</string>
        </property>
        <item>
         <property name="text">
          <string>Mouse</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Rat</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Monkey</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Human</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Other</string>
         </property>
        </item>
       </widget>
      </item>

      <item row="2" column="0">
       <widget class="QLabel" name="strain_label">
        <property name="text">
         <string>Strain/Genotype:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="strain_edit">
        <property name="placeholderText">
         <string>e.g., C57BL/6J, Wistar</string>
        </property>
       </widget>
      </item>

      <item row="3" column="0">
       <widget class="QLabel" name="sex_label">
        <property name="text">
         <string>Sex:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QComboBox" name="sex_combo">
        <item>
         <property name="text">
          <string>Male</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Female</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Unknown</string>
         </property>
        </item>
       </widget>
      </item>

      <item row="4" column="0">
       <widget class="QLabel" name="dob_label">
        <property name="text">
         <string>Date of Birth:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QDateEdit" name="dob_edit">
        <property name="calendarPopup">
         <bool>true</bool>
        </property>
        <property name="toolTip">
         <string>Select the subject's date of birth</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="physical_group">
     <property name="title">
      <string>Physical Characteristics</string>
     </property>
     <layout class="QFormLayout" name="physical_layout">

      <item row="0" column="0">
       <widget class="QLabel" name="weight_label">
        <property name="text">
         <string>Initial Weight:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QDoubleSpinBox" name="weight_spinbox">
        <property name="suffix">
         <string> g</string>
        </property>
        <property name="minimum">
         <double>0.0</double>
        </property>
        <property name="maximum">
         <double>1000.0</double>
        </property>
        <property name="singleStep">
         <double>0.1</double>
        </property>
        <property name="toolTip">
         <string>Initial weight in grams</string>
        </property>
       </widget>
      </item>

      <item row="1" column="0">
       <widget class="QLabel" name="identification_label">
        <property name="text">
         <string>Identification:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="identification_edit">
        <property name="placeholderText">
         <string>Ear tags, tattoos, microchip, etc.</string>
        </property>
        <property name="toolTip">
         <string>Any physical identification marks or methods</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="housing_group">     <property name="title">
      <string>Housing and Care</string>
     </property>
     <layout class="QFormLayout" name="housing_layout">

      <item row="0" column="0">
       <widget class="QLabel" name="housing_label">
        <property name="text">
         <string>Housing Location:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="housing_edit">
        <property name="placeholderText">
         <string>Room, rack, cage number</string>
        </property>
       </widget>
      </item>

      <item row="1" column="0">
       <widget class="QLabel" name="water_restriction_label">
        <property name="text">
         <string>Water Restricted:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QCheckBox" name="water_restriction_check">
        <property name="text">
         <string>Subject is on water restriction schedule</string>
        </property>
       </widget>
      </item>

      <item row="2" column="0">
       <widget class="QLabel" name="notes_label">
        <property name="text">
         <string>Notes:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QPlainTextEdit" name="notes_text">
        <property name="placeholderText">
         <string>Additional notes about the subject...</string>
        </property>
        <property name="maximumHeight">
         <number>80</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="vertical_spacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QLabel" name="required_note">
     <property name="text">
      <string>* Required fields</string>
     </property>
     <property name="styleSheet">
      <string>QLabel { color: #e74c3c; font-style: italic; }</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="button_box">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
     <property name="centerButtons">
      <bool>true</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>button_box</sender>
   <signal>accepted()</signal>
   <receiver>SubjectDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>554</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_box</sender>
   <signal>rejected()</signal>
   <receiver>SubjectDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>560</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
